misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -79,52 +102,137 @@ class CryptoModule(
  * Extension function for convenient encryption of [ByteString]s.
  * This function also makes sure that no extra copies of the plaintext data are kept in memory.
  */
+@Deprecated(
+    message = "Prefer to use encryptin methods that accept a Map<String, String> for encryption context",
+    replaceWith = ReplaceWith(
+        expression = "aead.encrypt(plaintext, encryptionContext)",
+        imports = ["misk.crypto.CryptoModule.encrypt"]))
 fun Aead.encrypt(plaintext: ByteString, aad: ByteArray? = null): ByteString {
   val plaintextBytes = plaintext.toByteArray()
   val encrypted = this.encrypt(plaintextBytes, aad)
   plaintextBytes.fill(0)
   return encrypted.toByteString()
 }
 
+/**
+ * Extension function for convenient encryption of [ByteString]s
+ * using a map of strings as encryption context.
+ * This function also makes sure that no extra copies of the plaintext data are kept in memory.
+ */
+fun Aead.encrypt(
+  plaintext: ByteString,
+  encryptionContext: Map<String, String>? = null
+) : ByteString {
+  val serializedEncryptionContext = encryptionContext?.let { serialize(it) }
+  val plaintextBytes = plaintext.toByteArray()
+  val encryptedByteString = this.encrypt(plaintextBytes, serializedEncryptionContext)
+  plaintextBytes.fill(0)
+  return CiphertextFormatter.serialize(encryptedByteString, encryptionContext)
```

Comment:
It looks like our ciphertext format wraps the Tink ciphertext format. Can we document what our ciphertext wire format looks like in a markdown file (e.g. [here is Tink's as an example](https://github.com/google/tink/blob/master/docs/WIRE-FORMAT.md)) to help for debugging? For example, it has been helpful to look for Tink's prefix byte, key id hint, and length in a hexdump of data to identify that it is Tink-encrypted data. A complicated example could document what our ciphertext+EncryptionContext format looks like for an envelope-encrypted Tink ciphertext.